### DISCUSSION TITLE
Jamulus 3.7.0 final is here!

### DISCUSSION THREAD

**Discussion by hoffie** [jamulussoftware/jamulus#1293] [Announcements]
We're excited to announce the availability of Jamulus 3.7.0!

This release re-organizes Server lists to make room for more servers:
- Default has been renamed to Any Genre 1.
- Any Genre has been renamed to Any Genre 2.
- Any Genre 3 has been added.
- Choral/Barbershop has been added.
- Classical/Folk/Choral has been renamed to Classical/Folk.

Besides that, numerous usability improvements, enhancements, bugfixes and optimizations have been integrated.
Please find all the details in the [Changelog](https://github.com/jamulussoftware/jamulus/releases/tag/r3_7_0).

Note: If you are using ASIO4ALL, we strongly suggest that you take a screenshot of your ASIO4ALL settings before upgrading.
The installer will try its best to migrate your settings, but it may fail to do so when using custom installation paths.

Downloads (primarily on Github, alternatively on [SourceForge](https://sourceforge.net/projects/llcon/files/latest/download)):
- [Windows](https://github.com/jamulussoftware/jamulus/releases/download/r3_7_0/jamulus_3.7.0_win.exe)
- [Mac](https://github.com/jamulussoftware/jamulus/releases/download/r3_7_0/jamulus_3.7.0_mac.dmg)
- [Debian/Ubuntu (amd64)](https://github.com/jamulussoftware/jamulus/releases/download/r3_7_0/jamulus_3.7.0_ubuntu_amd64.deb), alternative: [headless version](https://github.com/jamulussoftware/jamulus/releases/download/r3_7_0/jamulus_headless_3.7.0_ubuntu_amd64.deb)
- [Source code](https://github.com/jamulussoftware/jamulus/archive/r3_7_0.tar.gz)

This is the first release after the founder, developer and project lead of Jamulus, Volker Fischer (@corrados), [decided to leave the project](https://github.com/jamulussoftware/jamulus/discussions/928).
We want to use this opportunity to thank Volker again. Creating Jamulus and pushing it to the point where it is today was and is a huge accomplishment! Thanks, Volker!

Starting with this release, the project will be led by the Main Developers Team.

Thanks to everyone who did their part to make this release happen:
- Code contributors: @ann0see @atsampson @corrados @dakhubgit @dcorson-ticino-com @doloopuntil @freddii @geheimerEichkater @gilgongo @hoffie @jdrage @jerogee @jujudusud @mirabilos @mulyaj @nefarius2001 @npostavs @passing @pljones @ranfdev @softins @tormodvolden @vdellamea
- Application translators: @ann0see @corrados @dzpex @genesisproject2020 @hoffie @ignotus666 @jerogee @jose1711 @jujudusud @melcon @pljones @rolamos @SeeLook @Snayler @softins @trebmuh
- Website contributors/translators: @achim-grosse-oetringhaus @ann0see @dzpex @gegeweb @gilgongo @hoffie @ignotus666 @jdrage @mulyaj @trebmuh @vdellamea
- Main Developers: @ann0see @gilgongo @hoffie @pljones @softins
- ... and lots of people who brought new ideas or suggestions, guided their local colleagues or helped in various other ways!

If you have any questions or feedback, we encourage you to start a [Discussion on Github](https://github.com/jamulussoftware/jamulus/discussions/new).

**Comment by classicvictor** [comment]
👏🏼👏🏽👏🏾

**Reply by gegeweb** [reply]
Is there a source deb ?
I could be able to build the binary for armhf and arm64 (Raspberry Pi OS, buster an bullseye).

**Reply by ann0see** [reply]
No, unfortunately not. You can of course create one and add your scripts to this repo

**Comment by mirabilos** [comment]
Gérald Niel dixit:

>Is there a source deb ?

There will be one when I update the Debian package to 3.7.0.
I could probably do that in experimental if necessary, but not
in unstable right now as we’re preparing to release Debian 11
“bullseye” and necessary patches (see the other issue) go through
unstable currently.

I haven’t planned doing so at the moment (I’d rather see people
test the 3.6.2+dfsg1-2 we’ll be shipping, and my time is split
among many projects) but could be convinced if necessary.

bye,
//mirabilos
-- 
This space for rent.

https://paypal.me/mirabilos to support my work.

**Comment by wlcutler** [comment]
I clicked the windows button and downloaded the new version of Jamulus.
For the first time ever, I was given the message by my Edge browser:
jamulus_3.7.0_win.exe was blocked because it could harm your device. 
And the only option available was Delete. 

It has never given me that message before, and I have done quite a few downloads, including the 3.6.2 installer.
Would you be able to fix whatever it is that is giving the Edge browser the heebie-jeebies?

**Reply by ann0see** [reply]
Yes. That's SmartScreen and questionable design from Microsoft. You can open the installer by clicking on the 3 dots, afterwards on more information and then on something like run anyway. But they try to make it as hard as possible by hiding it. 

Fixing that can be done by:

1. zipping the exe file (ironically this warning will not come up if you download zip files)
2. Pay a tremendous amount of money

**Reply by npostavs** [reply]
Yeah, the thing to press is "Keep". I believe the "More Information" and then "Run Anyway" happens when you try to actually run the installer.
![edge-jamulus-blocked-download](https://user-images.githubusercontent.com/287742/111785282-c82e1c80-8892-11eb-9329-2824f9a83bfc.png)

**Reply by wlcutler** [reply]
Thank you for the replies. What concerns me is that Edge does not routinely do that. I have not seen that message before. So it's not so much "how do I run it anyway", as what is triggering this message and should I be believing it.

**Reply by npostavs** [reply]
I think after enough people report it as safe the warning goes away?

**Reply by ann0see** [reply]
See: https://docs.microsoft.com/en-us/windows/security/threat-protection/microsoft-defender-smartscreen/microsoft-defender-smartscreen-overview

> Checking downloaded files against a list of files that are well known and downloaded by many Windows users. If the file isn't on that list, Microsoft Defender SmartScreen shows a warning, advising caution.

**Reply by jerogee** [reply]
> So it's not so much "how to I run it anyway", as what is triggering this message and should I be believing it.

Microsoft uses a system of reputation: if an executable like this particular installer is opened by a certain number of users who do not report issues, then Microsoft gives it a green light. As developer, you can pay $699.00 USD for a EV Code Signing Certificate to "buy reputation".

**Reply by hsweet** [reply]
The Linux Foundation has recently https://sigstore.dev/ announced  " A non-profit, public good software signing & transparency service " https://sigstore.dev/  If I understand correctly, this should allow Jamulus to install on Macs and Windows without scarring away customers.

**Reply by ann0see** [reply]
That's great to hear! Do you have contact to them? Hopefully it'll support our automated build process.

**Reply by hoffie** [reply]
Yes, this sounds really great, but it seems we'll have to [wait some time](https://sigstore.dev/what_is_sigstore/). :(

> At present the WebPKI and client signing tooling is under prototype development and is not available for general use.

**Comment by wlcutler** [comment]
Almost OK. Thanks. 
I'm sure I didn't get that message from the original installation. Maybe I downloaded it from a different page?

**Reply by softins** [reply]
Versions up to 3.6.2 were primarily hosted on SourceForge. 3.7.0 primary host is GitHub, with SourceForge as an alternative. I wonder if SmartScreen looks at where the file was downloaded from, or just at a hash of the file? i.e. would it complain about the same file wherever it was downloaded from?

**Reply by jerogee** [reply]
@softins, it indeed might be that the domain it is downloaded from is also evaluated. @wlcutler, if this is the case you could try to download the installer from the same source that you used in the past? (and let us know if that makes a difference?)

Another consideration: this reputation build-up is specific to each executable/file, so will start at zero with each new version I'm afraid. (unless Microsoft also allows reputation to accumulate based on where the executables are published)   _If the old installation concerns a different version_, it can happen that when you downloaded the installler of the previous version it had already gathered enough reputation.

**Comment by wlcutler** [comment]
Thanks again for all this encouragement. I have installed it now from the Download link on SourceForge, got the same message as before, reported it as safe, saved it anyway, installed it anyway. 
![image](https://user-images.githubusercontent.com/80934297/111834094-55cc3500-88b0-11eb-9312-52a97a9a7785.png)

**Reply by ann0see** [reply]
Yes. You'll need to click on Show more. I hope we'll get around it soon (after more people have downloaded it).

**Reply by henkdegroot** [reply]
I have just downloaded the 3.7.0 exe and did not git a warning from Edge. I have to admit I did report it to microsoft as being safe and also select keep on this device. After this I was curious and downloaded from another windows device with Edge....and guess what ALL FINE....no blocking.

So it looks like it got better/easier now to download and install the update.

Happy Jamulus time!

**Comment by cellomike** [comment]
I've been a Jamulus user on Win10 since last May; it's been just terrific.  Downloading & installing new updates has been fine in the past, so updating to 3.7.0 should be no problem now on my Windows PC.  
However, recently I got a Raspberry Pi 400.  I'm a newbie on it; with help, was able to install Jamulus on it; runs very fast!  However, I'm not sure how to update the R Pi to the new version.  FYI, I was able to use a simple command-line prompt to update/install 3.6.2:  

         jamulus_update.sh r3_6_2 

Question: Is this still the command to use to update to 3.7.0 on the R Pi (substituting r3_7_0), or is there another command I need to use now for an update?  Any suggestions appreciated.  Thanks!

**Reply by ann0see** [reply]
Although I don't know the script you're using, I'd just try it. Not much can go wrong.

**Reply by wlcutler** [reply]
I know NOTHING about this, least of which what a Pi is, though a friend has told me. But the guy who runs our group's server on his Pi has just told me it was easy: "there is a big button on the Pi’s desktop that reads “Upgrade.” Though he did say he "ran into a bit of a snag when I tried to update the server, so I had to kill it and build a fresh one."

**Reply by cellomike** [reply]
Thanks, all, very much for your replies.  I have a Pi that has just the Pi OS and Jamulus on a mini SD card; no Jambox or "upgrade" option on the desktop.  I've gotten some suggestions on longer scripts; just not sure what to use.  My main concern is not disabling/blowing up current good Jamulus connectivity on the Pi while trying to update.

**Reply by ann0see** [reply]
I'm quite sure you can just follow the initial steps you used. Can you link the page which told you what to do or can you ask the person who set up the Raspberry Pi for you?

**Reply by cellomike** [reply]
Great, thanks.  I'll check with the person who set it up for me, then let you know.  Appreciate it!

**Comment by hsweet** [comment]
I have no idea as to the details, but I would be happy to help, if it is
not too far above my level of expertise.  I've gotten certs set up on
websites, which I think is similar.

I don't personally run into that as I use Linux locally on on my server,
but everyone else I play with runs into this
h

**Comment by ManfredSchenk** [comment]
I've already upgraded to the new release but now I'm msiing the Pan-controls above the faders. Are they still there or were they removed. I already searched the configuration but  I could find settings were they would appear.

**Reply by npostavs** [reply]
Your settings might have been reset, the pan controls only show up in "Stereo" or "Mono-in/Stereo-out" modes.

**Reply by ManfredSchenk** [reply]
Thank for your reply. I just found that it depends on the server I'm connected to. Perhaps I just didn't notice that with the previous version.

**Reply by hoffie** [reply]
> Your settings might have been reset, the pan controls only show up in "Stereo" or "Mono-in/Stereo-out" modes.

That would explain it, but I'd hope that settings would not reset arbitrarily. ;o

> I just found that it depends on the server I'm connected to.

Yes, this may be the reason. What version does the server with missing pan support run? It seems like it was [introduced](https://github.com/jamulussoftware/jamulus/blob/d854398080cc9613f6f2735828d1cd5cccfe2a97/ChangeLog#L391) in 3.5.4.

The UI [adapts](https://github.com/jamulussoftware/jamulus/blob/master/src/clientdlg.cpp#L843) accordingly.
https://github.com/jamulussoftware/jamulus/blob/master/ChangeLog#L391

**Comment by dakhubgit** [comment]
Noam Postavsky ***@***.***> writes:

> Your settings might have been reset, the pan controls only show up in
> "Stereo" or "Mono-in/Stereo-out" modes.

Which begs the question: "why"?  Even in "Mono" mode, the pan control
would affect the balance of a mixer strip from someone who sends stereo
signals.

-- 
David Kastrup

**Reply by softins** [reply]
It's because the panning is done in the server, not the client. In full "Mono" mode, there is only a single channel of mixed audio from the server to the client, which then gets sent by the client to both L and R.

In "Mono-in/Stereo-out" and full "Stereo" modes, there are two channels (L&R) of mixed audio from server to client. The server uses the pan setting for each channel when mixing L and R separately for the client.

**Comment by dakhubgit** [comment]
Tony Mountifield ***@***.***> writes:

> It's because the panning is done in the server, not the client. In
> full "Mono" mode, there is only a single channel of mixed audio from
> the server to the client, which then gets sent by the client to both L
> and R.

Sigh.  And this single channel is _different_ depending on the pan
position when the _source_ of that mixer strip has sent a stereo signal
to the server, even if the _recipient_ has pure mono output.

> In "Mono-in/Stereo-out" and full "Stereo" modes, there are two
> channels (L&R) of mixed audio from server to client. The server uses
> the pan setting for each channel when mixing L and R separately for
> the client.

-- 
David Kastrup

**Reply by gilgongo** [reply]
Can you explain what the practical effect of this issue gives rise to?

**Reply by softins** [reply]
> Sigh. 

Sighing doesn’t help; just explain and debate your point. We all have things to learn from each other.

> And this single channel is _different_ depending on the pan position when the _source_ of that mixer strip has sent a stereo signal to the server, even if the _recipient_ has pure mono output.

Ah, I understand what you’re saying, and it’s a possibility I hadn’t considered. Maybe Volker didn’t either, when implementing Pan. Or maybe he did.

I think what you are saying is that with a stereo input signal, and a mono output signal with Pan, if the Pan is set full left, the mono output only contains the left input, and if the Pan is set full right, the mono output only contains the right input. And of course, proportionally between those two extremes.

This is true, but of course, that received mono signal will still be of equal strength in both the recipient’s ears, and not spatially separated. That may certainly be useful in some contexts, but I think it is different from true panning. If its benefit could be demonstrated, we could certainly consider it.

What I understand true (volume-based) panning to be is where moving the pan control for an input hard to the left causes that input to appear only in the recipient’s left ear, and similarly hard to the right causes it to appear only in the right ear. It is this that gives the spatial separation, and can't be achieved with mono output.

I’m not sure, without studying the code, whether with a hard-left pan, for example, the recipient’s left ear receives any part of the sender’s right channel, or only their left.

And there is an open PR for a different kind of pan called delay panning, where the spatial effect is created not using volume differences, but rather small time delays between left and right of each source, according to the recipient’s pan setting for that source. Again, that relies on there being two channels to the recipient, otherwise it is potentially just a very short reverb effect.

**Comment by mirabilos** [comment]
Tony Mountifield dixit:

>I’m not sure, without studying the code, whether with a hard-left pan,
>for example, the recipient’s left ear receives any part of the sender’s
>right channel, or only their left.

I’d expect it to contain both. I understood panning to work kind of like
rotating your head, or moving the person in the room: their full output
is mixed and redistributed so that it “comes more from the left/right” in
my own stereo.

But I’m not a sound technician…

bye,
//mirabilos
-- 
  “Having a smoking section in a restaurant is like having
          a peeing section in a swimming pool.”
						-- Edward Burr

**Comment by dakhubgit** [comment]
Tony Mountifield ***@***.***> writes:

>> Sigh. 
>
> Sighing doesn’t help; just explain and debate your point. We all have
> things to learn from each other.

I already did and you ignored it, basically suggesting that I was so
clueless that it wasn't even worth considering what I wrote.

So I had to repeat it, wasting everybody's time.

>> And this single channel is _different_ depending on the pan position
>> when the _source_ of that mixer strip has sent a stereo signal to the
>> server, even if the _recipient_ has pure mono output.
>
> Ah, I understand what you’re saying, and it’s a possibility I hadn’t
> considered. Maybe Volker didn’t either, when implementing Pan. Or
> maybe he did.
>
> I think what you are saying is that with a stereo input signal, and a
> mono output signal with Pan, if the Pan is set full left, the mono
> output only contains the left input, and if the Pan is set full right,
> the mono output only contains the right input. And of course,
> proportionally between those two extremes.
>
> This is true, but of course, that received mono signal will still be
> of equal strength in both the recipient’s ears, and not spatially
> separated.

This is how a pan control on a stereo signal works on an actual physical
mixer.  If you wanted the same flexibility as two mono strips with fader
and pan on a stereo mixer, in addition to the volume fader you'd need
all of a balance control, a stereo width control, and a pan control.

On physical mixers all you get is the same pan control you have for a
mono signal, just that it doesn't get the same signal on the left and
the right but different signals.  If you use the mono output of the
mixer, this makes it work as a balance control.

> That may certainly be useful in some contexts, but I think it is
> different from true panning.

On a physical mixer, there is no such thing as "true panning".  If you
actually want a panned signal, you need "surround processing" like sox'
"earwax" filter does.

> If its benefit could be demonstrated, we could certainly consider it.

It has an effect, it has an obvious definition that agrees with what a
physical mixer does.  It has no net effect on mono signals on a mono
mix, but the Jamulus server doesn't even have the information "mono"
when you work with "mono-in/stereo-out": it receives two identical
channels.

> What I understand true (volume-based) panning to be is where moving
> the pan control for an input hard to the left causes that input to
> appear only in the recipient’s left ear, and similarly hard to the
> right causes it to appear only in the right ear. It is this that gives
> the spatial separation, and can't be achieved with mono output.

I have no idea what you call "true".  I am talking about the
well-established behavior of actual physical mixers.  It is not "true"
in any manner of the word, it is just how controls labeled like that and
looking like that work.

> I’m not sure, without studying the code, whether with a hard-left pan,
> for example, the recipient’s left ear receives any part of the
> sender’s right channel, or only their left.

Only the left.  That's how mixers implement it.  It works as a "pan"
(not with spatial impulse responses but by volume distribution) when
both left and right channel are fed the same signal.  When they are fed
different signals, they remain strictly separated and only their
relative volumes are adjusted.

> And there is an open PR for a different kind of pan called delay
> panning, where the spatial effect is created not using volume
> differences, but rather small time delays between left and right of
> each source, according to the recipient’s pan setting for that
> source. Again, that relies on there being two channels to the
> recipient, otherwise it is potentially just a very short reverb
> effect.

If you want this to be useful, there is not just time delays but a
mapping to binaural impulse responses.  Which includes both volume and
timing cues.  And is the kind of stuff you could add with plugin
support.

I have some wireless Sennheiser headphone where I can switch on a
"surround" effect and it essentially does that with a signal that has
just been panned by volume.

-- 
David Kastrup

**Reply by softins** [reply]
> > Sigh. Sighing doesn’t help; just explain and debate your point. We all have things to learn from each other.

> I already did and you ignored it, basically suggesting that I was so clueless that it wasn't even worth considering what I wrote.

Well I'm sorry you took it like that, which was not my intention. Closer to the truth would be that I failed to understand exactly what you were saying.

> So I had to repeat it, wasting everybody's time.

It's never a waste of time to make oneself clearer.

I'll give more thought to the rest of what you said.

> > And there is an open PR for a different kind of pan called delay panning, where the spatial effect is created not using volume differences, but rather small time delays between left and right of each source, according to the recipient’s pan setting for that source. Again, that relies on there being two channels to the recipient, otherwise it is potentially just a very short reverb effect.

> If you want this to be useful, there is not just time delays but a mapping to binaural impulse responses. Which includes both volume and timing cues. And is the kind of stuff you could add with plugin support. I have some wireless Sennheiser headphone where I can switch on a "surround" effect and it essentially does that with a signal that has just been panned by volume.

It's not my concept or implementation, but rather submitted by someone else. You can read about it, and comment if you wish, at #1151

**Comment by dakhubgit** [comment]
mirabilos ***@***.***> writes:

> Tony Mountifield dixit:
>
>>I’m not sure, without studying the code, whether with a hard-left pan,
>>for example, the recipient’s left ear receives any part of the sender’s
>>right channel, or only their left.
>
> I’d expect it to contain both. I understood panning to work kind of like
> rotating your head, or moving the person in the room: their full output
> is mixed and redistributed so that it “comes more from the left/right” in
> my own stereo.
>
> But I’m not a sound technician…

For those who do not want to believe me, I can check where the
respective pages with the circuit diagrams from the Mackie mixers I use
are.

Just as a hint: for professional analog audio equipment, super low
additional noise (and no additional delay) for basic functionality is of
the very essence, so complicated things like delay lines (which aren't
even readily implementable with analog circuitry which is why there are
things like coil reverbs from the time where neither suitable digital
circuitry nor discrete-time analog bucket delays were available) are not
in there.  They are separate devices if necessary.

-- 
David Kastrup

**Comment by animalglx** [comment]
Until version 3.7.0. it was easy to install but now it ain't work anymore.
In the past I used following commands (after downloading the jamulus-master.zip file):
qmake "CONFIG+=nosound" Jamulus.pro
make clean
make
sudo cp Jamulus.* /usr/local/bin 

Why isn't that working anymore?????????????

**Reply by npostavs** [reply]
> sudo cp Jamulus.* /usr/local/bin

I think you want `sudo make install` instead, but since you didn't say anything about the error messages it's hard to say why it doesn't work...

**Reply by hoffie** [reply]
Just a wild guess: You may be hitting #1304 (if your error message is about not finding `lrelease`). The workaround is to install `qttools5-dev-tools`. You could also try (untested) `CONFIG-=lrelease`.

**Reply by animalglx** [reply]
I ain't getting any error messages but Jamulus server remains version 3.6.2

**Reply by npostavs** [reply]
> I ain't getting any error messages but Jamulus server remains version 3.6.2

Okay, then I think the problem is that `sudo cp Jamulus.* /usr/local/bin` is installing `Jamulus.pro` which is the QMakefile, instead of the exectuable which is called `Jamulus`. `sudo make install` should work.

**Reply by animalglx** [reply]
Maybe I missed this message:"Project MESSAGE: building version "3.7.0dev-nogit" (intermediate without git repository)
What does it mean??

I'm using Ubuntu 18.04

**Reply by jerogee** [reply]
@animalglx , like @npostavs mentioned the proper way to install it system-wide is with `sudo make install`.

> I ain't getting any error messages but Jamulus server remains version 3.6.2

If you manually copied stuff in the past, first find out where the old one lives and remove it manually:
```bash
$ which Jamulus
/usr/bin/Jamulus
$ Jamulus --version
 *** Jamulus, Version 3.6.2
$ sudo rm -f /usr/bin/Jamulus
```
Then check again:
```bash
$ which Jamulus
/usr/local/bin/Jamulus
$ Jamulus --version
 *** Jamulus, Version 3.7.0
```
or if needed install again:
```bash
$ sudo make install
/usr/lib/qt5/bin/qmake -install qinstall -exe Jamulus /usr/local/bin/Jamulus
strip /usr/local/bin/Jamulus
/usr/lib/qt5/bin/qmake -install qinstall /root/jam/jamulus-r3_7_0/distributions/jamulus.desktop /usr/local/share/applications/jamulus.desktop
/usr/lib/qt5/bin/qmake -install qinstall /root/jam/jamulus-r3_7_0/distributions/jamulus.png /usr/local/share/icons/hicolor/512x512/apps/jamulus.png
```

**Reply by animalglx** [reply]
Thnx guys for being so polite to help me, I've found the problem, it's a bit embarrassing to tell.
I've used the cmd:"sudo cp Jamulus.* /usr/local/bin"  --  It should have been "sudo cp Jamulus* /usr/local/bin".

Now it's working fine.

**Reply by pljones** [reply]
If you're running a Debian/Ubuntu derived distribution on x64, you could use the supplied builds to avoid have issues like this.  The executable would then be `jamulus` (or `jamulus-headless` for the headless, nosound server) rather than `Jamulus`, though.

If you're on a Debian/Ubuntu distribution, you might even prefer the distribution supplied builds and update using your normal server updates.

(Other Linux distributions also supply Jamulus.)
